linkerd/io/src/lib.rs

```diff
@@ -50,86 +52,95 @@ impl PeerAddr for tokio::io::DuplexStream {
         Ok(([0, 0, 0, 0], 0).into())
     }
 }
-mod internal {
```

Comment:
Why do we need to change the visibility as a part of this? If this is necessary, can it be decoupled from the upgrade?